src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -822,7 +822,8 @@ private void CheckMethodGroup(BoundMethodGroup node, MethodSymbol method, bool p
             CheckReceiverIfField(node.ReceiverOpt);
             CheckReferenceToMethodIfLocalFunction(node, method);
 
-            if (method is null || method.RequiresInstanceReceiver)
+            if (method is null || method.RequiresInstanceReceiver ||
+                (method.IsExtensionMethod && node.SearchExtensionMethods))
```

Comment:
>method.IsExtensionMethod && node.SearchExtensionMethods [](http://example.com/codeflow?start=17&length=55)

This feels fragile, the method is not part of the BoundMethodGroup and the group doesn't control how the method is used. Rather than inventing new ways to infer the answer, let's use information explicitly stored in ```BoundConversion``` and ```BoundDelegateCreationExpression``` nodes, specifically the ```IsExtensionMethod``` property. This will also be consistent with how ```LocalRewriter``` does that. Alternatively, we could always visit the receiver. Could any harm come from visiting a bound type, for example?

Reply:
Visiting unconditionally fails the following:

```csharp
using System;
class Program
{
    static void Main()
    {
        static void F1<T>()
        {
            Console.WriteLine(typeof(T));
        }
        static void F2()
        {
            F1<int>();
            Action a = F1<string>;
            a();
        }
        F2();
    }
}
```

The receiver here is `this`, but it's not really captured. I'm not sure though why the receiver is non-null value.

Reply:
> The receiver here is this, but it's not really captured. I'm not sure though why the receiver is non-null value.

It looks like unconditional visit wouldn't work. I forgot that binder synthesizes `this` receiver in cases like that too.